src/proxy/topic/TopicProxy.ts

```diff
@@ -0,0 +1,123 @@
+import {ITopic} from '../ITopic';
+import {Message, MessageListener} from '../MessageListener';
+import {AddressImpl, HazelcastError, UUID} from '../../core';
+import {TopicAddMessageListenerCodec} from '../../codec/TopicAddMessageListenerCodec';
+import {TopicRemoveMessageListenerCodec} from '../../codec/TopicRemoveMessageListenerCodec';
+import { ListenerMessageCodec} from '../../listener/ListenerMessageCodec';
+import {ClientMessage} from '../../protocol/ClientMessage';
+import {assertNotNull} from '../../util/Util';
+import {PartitionSpecificProxy} from '../PartitionSpecificProxy';
+import {TopicPublishCodec} from '../../codec/TopicPublishCodec';
+import Long = require('long');
+import {Data} from '../../serialization';
+import {ClientConfig, ClientConfigImpl} from '../../config';
+import {ProxyManager} from '../ProxyManager';
+import {PartitionService} from '../../PartitionService';
+import {InvocationService} from '../../invocation/InvocationService';
+import {SerializationService} from '../../serialization/SerializationService';
+import {ListenerService} from '../../listener/ListenerService';
+import {ClusterService} from '../../invocation/ClusterService';
+import {ConnectionRegistry} from '../../network/ConnectionRegistry';
+import {SchemaService} from '../../serialization/compact/SchemaService';
+import {Connection} from '../../network/Connection';
+import {TopicOverloadPolicy} from '../TopicOverloadPolicy';
+
+export class TopicProxy<E> extends PartitionSpecificProxy implements ITopic<E> {
+
+
+    private readonly overloadPolicy: TopicOverloadPolicy;
+    private readonly localAddress: AddressImpl;
+
+
+    constructor(
+        serviceName: string,
+        name: string,
+        clientConfig: ClientConfig,
+        proxyManager: ProxyManager,
+        partitionService: PartitionService,
+        invocationService: InvocationService,
+        serializationService: SerializationService,
+        listenerService: ListenerService,
+        clusterService: ClusterService,
+        connectionRegistry: ConnectionRegistry,
+        schemaService: SchemaService
+    ) {
+        super(
+            serviceName,
+            name,
+            proxyManager,
+            partitionService,
+            invocationService,
+            serializationService,
+            listenerService,
+            clusterService,
+            connectionRegistry,
+            schemaService
+        );
+        const connection: Connection = this.connectionRegistry.getRandomConnection();
+        this.localAddress = connection != null ? connection.getLocalAddress() : null;
+        const config = (clientConfig as ClientConfigImpl).getReliableTopicConfig(name);
+        this.overloadPolicy = config.overloadPolicy;
+    }
+
+    publish(message: E): Promise<void> {
```

Comment:
This method should be like this: 

```js
publish(message: E): Promise<void> {
        assertNotNull(message);

        const messageData = this.toData(message);
        return this.encodeInvoke(TopicPublishCodec, () => {}, messageData);
    }
```

Can you write some tests and try this if the tests pass? 

The explanation: You were using the partition id of the message. The idea is that we need to use partition id of the topic itself calculated using the proxy name because Topic is partitioned according to the name of the topic, not the data. You can see the details of how it is done in PartitionSpecificProxy class. That's why we use encodeInvoke now.

Reply:
I commented below about tests